src/drawing/DrawingList.tsx

```diff
@@ -21,6 +21,12 @@ export function filterDrawing({ target: { path_groups: pathGroups } }: Annotatio
     );
 }
 
+export function isSafari(): boolean {
+    const { userAgent } = navigator;
+    // In Chrome browser, the agent string contains Safari and Chrome but in Safari there is only Safari
+    return userAgent.indexOf('Safari/') > 0 && userAgent.indexOf('Chrome/') < 0;
```

Comment:
We should avoid user agent detection and browser-specific logic whenever possible. Please try to find a method that allows a universal solution.

Reply:
Could you describe why we should be avoiding user agent detection?

I've looked around at some of the other codebases that we own and it looks like in each browser-detection logic there's some variation of `userAgent` that is being used.

I looked at some npm libraries as well and they also seem to be relying on `userAgent` to detect the current browser.

Will keep looking for a suitable alternative.

Reply:
It's pretty common advice to avoid the practice ([1](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent), [2](https://css-tricks.com/freezing-user-agent-strings/), probably more) due to its fragility. Google has announced user agent strings will be frozen and/or removed in coming versions of Chrome, for example.

Examples of a less than ideal practice in other codebases shouldn't be enough to justify it in ours, especially since this codebase currently doesn't use it anywhere else today. We can discuss more offline, as well, just let me know.